subsys/bluetooth/host/id.c

```diff
@@ -1307,6 +1303,32 @@ int bt_id_delete(uint8_t id)
 	return 0;
 }
 
+
+int bt_id_delete(uint8_t id)
+{
+	if (id == BT_ID_DEFAULT || id >= bt_dev.id_count) {
+		return -EINVAL;
+	}
+
+	return id_delete(id);
+}
+
+int bt_id_delete_all(void)
+{
+	for (uint8_t id = 0U; id < bt_dev.id_count; id++) {
+		int err;
+
+		err = id_delete(id);
+		if (err) {
+			return err;
```

Comment:
You must ignore `err == -EALREADY`. There may be fragmentation/holes in `bt_dev.id_addr[0:bt_dev.id_count]` if an identity has been deleted before.

I recommend adding a few `id_create` and `id_delete` to the bablesim test. Add two `id_create`, and `id_delete` the first one you created.

Reply:
You make a good point

Reply:
Not sure this works too well in the babblesim test, a unit test would be ideal.